src/rttr/variant.h

```diff
@@ -423,6 +428,32 @@ class RTTR_API variant
          */
         bool is_sequential_container() const;
 
+        /*!
+         * \brief Returns a reference to the containing value as type \p T.
+         *
+         * \code{.cpp}
+         *  struct custom_type
+         *  {
+         *     //...
+         *  };
+         *
+         *  variant var = custom_type{};
+         *  if (var.is_type<custom_type>())                         // yields to true
+         *      custom_type& value = var.get_value<custom_type>();  // extracts the value by reference
+         * \endcode
+         *
+         * \remark Only call this method when it is possible to return the containing value as the given type \p T.
+         *         Use therefore the method \ref is_type().
+         *         Otherwise the call leads to undefined behaviour.
+         *         Also make sure you don't clean this variant, when you still hold a reference to the containing value.
+         *
+         * \see is_type()
+         *
+         * \return A reference to the stored value.
+         */
+        template<typename T>
+        T& get_value();
```

Comment:
I like that one. So such a functionality was not yet there. Its up to the user to call that with the right type.